tools/build/cross-build/include/common/sys/cdefs.h

```diff
@@ -116,15 +116,33 @@
 #define	__predict_false(exp)    __builtin_expect((exp), 0)
 #endif
 
-#ifndef __weak_reference
 #ifdef __ELF__
+#ifndef __weak_reference
 #define	__weak_reference(sym,alias)	\
 	__asm__(".weak " #alias);	\
 	__asm__(".equ "  #alias ", " #sym)
+#endif
+#ifndef __sym_compat
+#define	__sym_compat(sym,impl,verid)	\
+	__asm__(".symver " #impl ", " #sym "@" #verid)
+#endif
+#ifndef __sym_default
+#define	__sym_default(sym,impl,verid)	\
+	__asm__(".symver " #impl ", " #sym "@@@" #verid)
+#endif
+#ifndef __warn_references
+#define	__warn_references(sym,msg)	\
+	__asm__(".section .gnu.warning." #sym);	\
+	__asm__(".asciz \"" msg "\"");	\
+	__asm__(".previous")
+#endif
 #else
 #define __weak_reference(sym,alias)	\
     static int alias() __attribute__ ((weakref (#sym)));
```

Comment:
Would this not be better as `#pragma weak alias = sym`? I don't think your current solution even actually provides the semantics wanted by `__weak_reference`, namely to export the symbol (`weakref` requires it to be static) as weak. The pragma is old and supported by both GCC and Clang. Quite frankly FreeBSD should move to this for its own cdefs.h anyway :) it produces basically-identical code on Clang for ELF (.weak/.set rather than .weak/.equ), and  has the advantage that it also emits the correct .weak_reference for Mach-O.

Reply:
*sigh*, right, #pragma doesn't work in the body of #define... :(

Reply:
I just copied this from the main freebsd cdefs.h. but I guess I can just keep it empty since we don't really need it to work properly.

Reply:
There's also _Pragma that should work in macros

Reply:
Ah, yes, good, we should be able to rely on C99 and C++11 these days. You want:

```c
#define __do_pragma(x) _Pragma(#x)
#define __weak_reference(sym, alias) __do_pragma(weak alias = sym)
```
You can't just do `#define __weak_reference(sym, alias) _Pragma("weak "#alias" = "#sym)` as one would naively expect because `_Pragma` wants a single string literal and adjacent ones aren't implicitly concatenated.